django/api/services/spreadsheet_uploader.py

```diff
@@ -101,52 +102,45 @@ def transform_data(
-                                "Rows": [],
-                                "Severity": "Error"
-                            }
-                        errors_and_warnings[column]["Incorrect Date Format"]["Rows"].append(index + 1)
-
-            if expected_type in [int, float, Decimal] and value is not None and pd.notna(value) and value != '':
-                value = str(value).replace(',', '').strip()
-                try:
-                    if expected_type == int:
-                        row_dict[column] = int(float(value))
-                    elif expected_type == Decimal:
-                        row_dict[column] = Decimal(value).quantize(Decimal("0.01"), rounding=ROUND_HALF_UP)
-                    else:
-                        row_dict[column] = float(value)
-                except ValueError:
-                    if column not in errors_and_warnings:
-                        errors_and_warnings[column] = {}
-                    if "Incorrect Type" not in errors_and_warnings[column]:
-                        errors_and_warnings[column]["Incorrect Type"] = {
-                            "Expected Type": f"The following rows contained types for the column {column}. Expected {type_to_string.get(expected_type, str(expected_type))}",
-                            "Rows": [],
-                            "Severity": "Error"
-                        }
-                    errors_and_warnings[column]["Incorrect Type"]["Rows"].append(index + 1)
-
-            # Check if expected_type is valid before using isinstance
-            elif expected_type is not None and isinstance(expected_type, type) and not isinstance(row_dict[column], expected_type) and value != "":
-                if column not in errors_and_warnings:
-                    errors_and_warnings[column] = {}
-                if "Incorrect Type" not in errors_and_warnings[column]:
-                    errors_and_warnings[column]["Incorrect Type"] = {
-                        "Expected Type": f"The following rows contained types for the column {column}. Expected {type_to_string.get(expected_type, str(expected_type))}",
-                        "Rows": [],
-                        "Severity": "Error"
-                    }
-                errors_and_warnings[column]["Incorrect Type"]["Rows"].append(index + 1)
+                else:
+                    if expected_type:
+                        try:
+                            if expected_type == int:
+                                row_dict[column] = int(float(value))
+                            elif expected_type == float:
+                                row_dict[column] = float(value)
+                            elif expected_type == Decimal:
+                                row_dict[column] = Decimal(value).quantize(
+                                    Decimal("0.01"), rounding=ROUND_HALF_UP
+                                )
+                            elif expected_type is real_datetime:
```

Comment:
We're using `==` to check equality in the other cases above, but using `is` here; `is` checks to see if two references refer to the same thing in memory, which I think may be the case here even though `expected_type` and `date` are from two different imports in two different modules, but I think our intention is to check from logical equality, so probably better to use `==`.